Task3/Project/DataControl/Services/CsvFileService.cs

```diff
@@ -1,53 +1,301 @@
-using DataControl.Interfaces;
+using System.IO;
+using System.Collections.Generic;
 
 namespace DataControl.Services
 {
-    public class CsvFileService : IDataAccessService
+    /// <summary>
+    /// Represents service for work with csv files.
+    /// </summary>
+    public class CsvFileService : Interfaces.IDataAccessService
     {
+        // FIELDS
+        private TaxiDriver.Driver driver;
+        private string message;
+        private FileConfiguration fileConfiguration;
+        private System.Random rand;
+
+        // PROPERTIES
+        /// <summary>
+        /// Current driver.
+        /// </summary>
         public TaxiDriver.Driver Driver
         {
             get
             {
-                throw new System.NotImplementedException();
+                return driver;
             }
         }
-
+        /// <summary>
+        /// Some special information about exceptional service state.
+        /// </summary>
         public string Message
         {
             get
             {
-                throw new System.NotImplementedException();
+                return message;
             }
         }
 
+        // CONSTRUCTORS
+        /// <summary>
+        /// Basic constructors without parameters.
+        /// </summary>
+        public CsvFileService()
+        {
+            rand = new System.Random();
+        }
+
+        // METHODS
+        private void CheckFileConfiguration()
+        {
+            if (fileConfiguration == null)
+            {
+                throw new System.ArgumentNullException("The configuration isn`t setted up.");
+            }
+        }
+
+        private string[] GetRandomSplittedLine(string path)
+        {
+            string[] lines = File.ReadAllLines(path);
+            int index = rand.Next(0, lines.Length);
+

```

Comment:
could be optimized

Reply:
dont read all file, file could be big, and you dont need all information from it otherwise
read line by line. and random the chance to choose current line. if file ended start from begining. repeat it while line will not be choosed.